### PR TITLE
Calculate gutter width less often 

### PR DESCRIPTION
**Pull request by lewis6991**
Prevents glitchyness when editing at the end of the line when there is context displayed

### CODE REVIEW

File: lua/treesitter-context.lua [hunk 1]
```diff
@@ -362,15 +350,18 @@ function M.close()
   winid = nil
 end
 
-function M.open()
+function M.open(update_gutter_with)
```
Comment (romgrk) [inline lua/treesitter-context.lua:353]: typo `with` => `width`

File: lua/treesitter-context.lua [hunk 2]
```diff
@@ -469,8 +460,8 @@ function M.open()
           -- we replace "\n" with " "
           offset = intended_start_row
           -- Add the length of each precending lines
-          for i = 1, intended_start_row do
-            offset = offset + #lines[i] - indents[i]
```
Comment (romgrk) [inline lua/treesitter-context.lua:473]: Not 100% sure of the context here but why is this removed?

**Review by romgrk — COMMENTED** [review]
Thanks for the cleanup :] See two comments but otherwise LGTM

File: lua/treesitter-context.lua (hunk 2 above)
Comment (lewis6991) [reply lua/treesitter-context.lua:473]: line isn't defined in this context. Not sure why this hasn't caused issues, but it is neither a global or a local.

Comment (lewis6991) [reply lua/treesitter-context.lua:473]: Should maybe `lines` -> `context_lines` ?

Comment (romgrk) [reply lua/treesitter-context.lua:473]: My bad, it should be defined like this:

```diff
diff --git a/lua/treesitter-context.lua b/lua/treesitter-context.lua
index 4c21f10..fec6af8 100644
--- a/lua/treesitter-context.lua
+++ b/lua/treesitter-context.lua
@@ -418,6 +418,7 @@ function M.open(update_gutter_with)
     local current_node = context_nodes[i]
     local range = context_ranges[i]
     local indents = context_indents[i]
+    local lines = context_lines[i]
 
     local start_row = range[1]
     local start_col = range[2]
```

Revert the removal, I'll work on it this afternoon (I promise), I'll merge as soon as that and the typo are done, works fine for the rest.

Comment (lewis6991) [reply lua/treesitter-context.lua:473]: How about the latest version where I've just commented the line and added a FIXME?

Otherwise I'll revert.

Comment (romgrk) [reply lua/treesitter-context.lua:473]: Yep, good. Thanks

Comment (lewis6991) [reply lua/treesitter-context.lua:473]: Done
